core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala

```diff
@@ -156,47 +165,96 @@ object ConsumerGroupCommand {
       val props = if (opts.options.has(opts.commandConfigOpt)) Utils.loadProps(opts.options.valueOf(opts.commandConfigOpt)) else new Properties()
       val channelSocketTimeoutMs = props.getProperty("channelSocketTimeoutMs", "600").toInt
       val channelRetryBackoffMs = props.getProperty("channelRetryBackoffMsOpt", "300").toInt
+      if (!zkUtils.getConsumerGroups().contains(group)) {
+        println("The consumer group provided does not exist")
+        return
+      }
       val topics = zkUtils.getTopicsByConsumerGroup(group)
       if (topics.isEmpty)
         println("No topic available for consumer group provided")
       printDescribeHeader()
-      topics.foreach(topic => describeTopic(group, topic, channelSocketTimeoutMs, channelRetryBackoffMs))
-    }
 
-    private def describeTopic(group: String,
-                              topic: String,
-                              channelSocketTimeoutMs: Int,
-                              channelRetryBackoffMs: Int) {
-      val topicPartitions = getTopicPartitions(topic)
-      val groupDirs = new ZKGroupTopicDirs(group, topic)
-      val ownerByTopicPartition = topicPartitions.flatMap { topicPartition =>
-        zkUtils.readDataMaybeNull(groupDirs.consumerOwnerDir + "/" + topicPartition.partition)._1.map { owner =>
-          topicPartition -> owner
+      val topicPartitions = getTopicsPartitions(topics)
+      var groupMemberIds = zkUtils.getConsumersInGroup(group)
+
+      // mapping of topic partition -> ephemeral owner
+      val eOwnerByTopicPartition = mutable.Map[TopicAndPartition, Long]()
+      topicPartitions.foreach { topicPartition =>
+        val data = zkUtils.readDataMaybeNull(new ZKGroupTopicDirs(group, topicPartition.topic).consumerOwnerDir + "/" + topicPartition.partition)
+        val owner = data._1
+        if (!owner.isEmpty) {
+          val eOwner = data._2.getEphemeralOwner()
+          eOwnerByTopicPartition.put(topicPartition, eOwner)
```

Comment:
@junrao I was wondering what your opinion is about the response above to your comment. This PR has gone through a few more rounds of reviews thanks to @hachikuji and this currently is the only outstanding item. Thanks in advance for looking into this.